src/Voron/StorageEnvironment.cs

```diff
@@ -38,8 +38,54 @@ namespace Voron
 {
     public delegate bool UpgraderDelegate(SchemaUpgradeTransactions transactions, int currentVersion, out int versionAfterUpgrade);
 
-    public class StorageEnvironment : IDisposable
+
+    public class StorageEnvironmentSynchronization
     {
+        public struct SynchronizationSection : IDisposable
+        {
+            private readonly SemaphoreSlim _context;
+
+            public SynchronizationSection(SemaphoreSlim context, CancellationToken cancellationToken)
+            {
+                this._context = context;
+                if (this._context != null)
+                {
+                    this._context.Wait(cancellationToken);
+                    cancellationToken.ThrowIfCancellationRequested();
+                }                    
+            }
+
+            void IDisposable.Dispose()
+            {
+                if (this._context != null)
+                    _context.Release();
+            }
+        }
+
+        private readonly SemaphoreSlim _writeSemaphore;
+        public readonly long MaxWriteSize;
```

Comment:
rename to `MaxWriteSizeInBytes` ? Maybe leave it as `Size` ?

Reply:
Like `MaxWriteSizeInBytes` more, because it is more explicit.

Reply:
https://github.com/ravendb/ravendb/blob/v5.1/src/Sparrow/Size.cs

Reply:
Ahhhh you mean to use the `Size` type... I wasn't understanding... roger that